coalib/results/TextRange.py

```diff
@@ -89,7 +89,35 @@ def end(self):
         return self._end
 
     def overlaps(self, other):
-        return self.start <= other.end and self.end >= other.start
+        first = False  # check for left boundary overlap
+        # self.start <= other.end
+
+        second = False  # check for right boundary overlap
+        # self.end >= other.start
+
+        if None not in (self.start.line, other.end.line):
+            if self.start.line == other.end.line:
+                if None in (self.start.column, other.end.column):
+                    first = True  # if column = None, consider whole line
+                else:
+                    first = self.start.column <= other.end.column
```

Comment:
It is split up so that I could comment on the if part. So, its just for the sake of clarity.